Update Teleport Connect test plan

ravicious opened the pull request:
TODO:

- [x] Passwordless
- [x] Graceful cert handling
- [x] K8s
- [x] File transfer

[Rendered version](https://github.com/gravitational/teleport/blob/ravicious/update-testplan/.github/ISSUE_TEMPLATE/testplan.md#teleport-connect)

ravicious committed:
Add items done during v10 test plan

ravicious committed:
Add access requests test plan items

Co-authored-by: Michael <michael.myers@goteleport.com>

gzdunek committed:
Add kubernetes and file transfer test items

gzdunek committed:
Improve wording

ravicious commented on .github/ISSUE_TEMPLATE/testplan.md:
```diff
@@ -1352,13 +1377,33 @@ With a default Postgres DB instance, a Teleport instance configured with DB acce
+      - [ ] Stop a leaf cluster.
+      - [ ] Disconnect your device from the internet.
+  - DocumentGateway
+    - Create a db connection tab for a given database. Then remove access to that db for that user.
```
Scan the list again and add missing checkboxes.

ravicious commented on .github/ISSUE_TEMPLATE/testplan.md:
```diff
@@ -1352,13 +1377,33 @@ With a default Postgres DB instance, a Teleport instance configured with DB acce
+  - DocumentGateway
+    - Create a db connection tab for a given database. Then remove access to that db for that user.
+      Go back to Connect and:
+      - Change database name in the tab. This should still work.
+      - Change database port. This should result in an error.
```
This was written when I was still thinking that restarting a gateway is going to refetch the certs. Restarting a gateway doesn't update certs in any way, so both actions should work.

Look up the test plan for v11 to copy the updated test plan items for this section.

zmb3 commented:
@ravicious can we get this merged in the next week or two so it's ready for the v12 test plan?

ravicious commented:
> @ravicious can we get this merged in the next week or two so it's ready for the v12 test plan?

Sure; there were always more important things in the pipeline than updating the test plan so I've been postponing this since v11. 🙊

ravicious committed:
Fix indentation

ravicious committed:
Add missing checkboxes

ravicious committed:
Move gateway-specific items from cluster resources to DocumentGateway

ravicious committed:
Update DocumentGateway items related to changing port/db name

Changing the port does not refetch the certs so it shouldn't throw an error.

ravicious committed:
Remove duplicated item

The "State restoration from disk" section already includes an item for
that (next to last item).

ravicious committed:
Fix grammar

ravicious committed:
Move two items from "Cluster resources" to new "Tabs" section

ravicious committed:
Account for server-side fetching in DocumentCluster items

ravicious committed:
Add passwordless to "Auth methods" section

ravicious committed:
Merge branch 'master' into ravicious/update-testplan

ravicious committed:
Update items related to refreshing certs

ravicious committed:
Merge branch 'master' into ravicious/update-testplan

ravicious committed:
Merge branch 'master' into ravicious/update-testplan
